httpclient-okhttp/src/test/java/io/fabric8/kubernetes/client/okhttp/OkHttpClientFactoryTest.java

```diff
@@ -53,4 +56,21 @@ void shouldNotRespectMaxRequestsPerHost() {
     assertEquals(Integer.MAX_VALUE, client.getOkHttpClient().dispatcher().getMaxRequestsPerHost());
   }
 
+  @Test
+  void additionalConfigCanOverrideDispatcher() {
+    final Dispatcher dispatcher = new Dispatcher();
```

Comment:
Overriding completely via additionalConfig does preserve the legacy behavior.  Not sure if it needs an explicit test, but the intent of #4797 would be to override the initDispatcher:

> To get the best of both worlds, i.e. both the default Dispatcher configuration from https://github.com/fabric8io/kubernetes-client/pull/4687 and a daemon thread pool

Reply:
I didn't add a test for the initDispatcher override, but this fix allows for both. You can either override initDispatcher, or use additionalConfig.

I can add that test too. Not sure if something else is expected besides that